core/src/main/java/org/elasticsearch/action/support/replication/TransportWriteAction.java

```diff
@@ -124,7 +138,11 @@ public synchronized void respond(ActionListener<Response> listener) {
         protected void respondIfPossible(Exception ex) {
             if (finishedAsyncActions && listener != null) {
                 if (ex == null) {
-                    super.respond(listener);
+                    if (finalResponseIfSuccessful != null) {
```

Comment:
This was a leftover from me trying something out, I've removed it in 56cc63f612e5281555d5da1202dc9a44748eceea